From 4137914f6648cdcf0fcea63bd08e4a3ec8e1e39b Mon Sep 17 00:00:00 2001
From: Naushir Patuck <naush@raspberrypi.com>
Date: Fri, 31 Mar 2023 14:56:49 +0100
Subject: [PATCH] dtoverlays: Add link-frequency parameter for the Sony
 IMX708 sensor

Add a parameter to change the sensor device CSI-2 link frequency to
one of the following values: 450Mhz (default), 447Mhz, or 453Mhz.

Signed-off-by: Naushir Patuck <naush@raspberrypi.com>
---
 arch/arm/boot/dts/overlays/README             | 2 ++
 arch/arm/boot/dts/overlays/imx708-overlay.dts | 1 +
 2 files changed, 3 insertions(+)

--- a/arch/arm/boot/dts/overlays/README
+++ b/arch/arm/boot/dts/overlays/README
@@ -2417,6 +2417,8 @@ Params: rotation                Mounting
                                 configuring the sensor (default on)
         cam0                    Adopt the default configuration for CAM0 on a
                                 Compute Module (CSI0, i2c_vc, and cam0_reg).
+        link-frequency          Allowable link frequency values to use in Hz:
+                                450000000 (default), 447000000, 453000000.
 
 
 Name:   iqaudio-codec
--- a/arch/arm/boot/dts/overlays/imx708-overlay.dts
+++ b/arch/arm/boot/dts/overlays/imx708-overlay.dts
@@ -88,6 +88,7 @@
 		       <&vcm_node>, "VDD-supply:0=",<&cam0_reg>;
 		vcm = <&vcm_node>, "status",
 		      <0>, "=4";
+		link-frequency = <&cam_endpoint>,"link-frequencies#0";
 	};
 };
 
